components/base/server.go

```diff
@@ -162,6 +168,9 @@ func (s *serviceServer) GetGeometries(ctx context.Context, req *commonpb.GetGeom
 	if err != nil {
 		return nil, err
 	}
+	if geometries == nil {
+		return nil, ErrGeometriesNil(req.GetName())
+	}
 	return &commonpb.GetGeometriesResponse{Geometries: spatialmath.NewGeometriesToProto(geometries)}, nil
```

Comment:
@cheukt `NewGeometriesToProto` doesn't return an error, so I can add an error return, but that just requires me to change everywhere else it's used. it's just a couple components, but feels more breaking than just adding a nil check before we even get there. thoughts on leaving the geometries implementation at the client/server level?